Automattic/wp-calypso
# profileLinks: migrate to `@tanstack/react-query`

## Proposed Changes

This PR covers a few migrations with `<ProfileLinks />`

- Migrate data fetching and mutations from our legacy data layer and Redux to using `@tanstack/react-query`. We follow the guideline to place the hooks alongside where they're used and therefore those can all be found in `client/me/profile-links/data`.
- Migrate any components to TypeScript
- Migrate any class components to function components

Previously we had a few different profile links related directories in `client/me`. I rearranged them to all be in `client/me/profile-links`. They're not used anywhere else and for me it makes more sense to have them contained in a single directory.

While generally profile links should behave exactly the same we introduce some UX improvements:

- show `busy` states in case a mutation is running (when adding or removing profile links)
- when adding links we now account for the case where we could have multiple errors and success when adding links from WordPress.com and show an error or success notice per link
- show notices when removing profile links

## Testing Instructions

* Go to `/me` and smoke test the profile links section
* Try adding and removing links, verify those work correctly
* Try error cases (adding the same link twice) or block network requests (via your browser devtools)
* Verify unit tests pass and there are no type errors

If you're wondering where those profile links show up after adding them, go to `https://gravatar.com/:wpcom-username` and you should see the list there.

## Pre-merge Checklist
